spec/models/spree/order_spec.rb

```diff
@@ -1357,4 +1357,48 @@ def advance_to_delivery_state(order)
       end
     end
   end
+
+  describe 'editing a complete order, with two line items' do
+    let!(:payment_method) { create(:payment_method) }
+    let!(:shipping_method) { create(:shipping_method) }
+    let(:order) { create(:order_ready_to_ship, line_items_count: 2) }
+    let(:item_num) { order.line_items.length }
+    let(:item1_price) { order.line_items.first.price }
+    let(:item2_price) { order.line_items.second.price }
+    let(:customer_balance) { order.total - order.payment_total }
+
+    context 'when an item is removed' do
+      before do
+        order.line_items.first.destroy
+        order.update!
+      end
+
+      it 'updates the payment state of the order' do
+        expect(item_num).to eq 1
```

Comment:
IMO this is redundant and it's actually testing Rails itself. If you look at it, we're checking that destroy removes the line item record and that IMO falls out of the responsibility of this test.